apis/kueue/v1beta1/admissioncheck_types.go

```diff
@@ -25,16 +25,14 @@ type CheckState string
 const (
 	// CheckStateRetry means that the check cannot pass at this moment, back off (possibly
 	// allowing other to try, unblock quota) and retry.
-	// A workload having at least one check in the state,
-	// will be evicted if admitted will not be considered
-	// for admission.
+	// A workload having at least one check in the state, will be evicted if admitted and
```

Comment:
```suggestion
	// A workload having at least one check in the state is requeued. An admitted workload
	// is evicted before requeuing.
```
I guess it still is considered for admission, once it gets the quota reservation again?

Reply:
Ok, but it still can be considered for admission if the admission check is flipped. I think this is a purpose of Retry to re-consider it for admission, contrary to Reject.

Reply:
This PR only changes the comment to better describe the behavior.

Reply:
Ok, maybe then this is enough:
`// An admitted workload having at least one check in the state is evicted.`

Reply:
Why make it incomplete. 

The changed I've done here is 

```diff
- A workload having at least one check in the state, will be evicted if admitted will not be considered for admission.
+ A workload having at least one check in the state, will be evicted if admitted and not be considered for admission.
```

Reply:
sounds acceptable, but let me defer to the API reviewers @alculquicondor @tenzen-y